lib/editor/components/ExceptionDateRange.js

```diff
@@ -0,0 +1,159 @@
+// @flow
+
+import React, {Component} from 'react'
+import {DropdownButton, MenuItem} from 'react-bootstrap'
+import DateTimeField from 'react-bootstrap-datetimepicker'
+import moment from 'moment'
+
+import {updateActiveGtfsEntity} from '../actions/active'
+import {modifyRangeOfDates, updateDates} from '../../common/util/exceptions'
+import type {ScheduleException, ScheduleExceptionDateRange} from '../../types'
+
+import {inputStyleProps} from './ExceptionDate'
+
+type Props = {
+  activeComponent: string,
+  activeEntity: ScheduleException,
+  index: number,
+  isInvalid: boolean,
+  range: ScheduleExceptionDateRange,
+  updateActiveGtfsEntity: typeof updateActiveGtfsEntity
+}
+
+export default class ExceptionDateRange extends Component<Props> {
+  // This is kept as an arrow function in order to access the component's `this`
+  _onRangeUpdate = (millis: number, isStartDate: boolean) => {
+    const {activeComponent, activeEntity, range, updateActiveGtfsEntity} = this.props
+    const {endDate, startDate} = range
+
+    const newRangeBoundary = moment(+millis)
+    const startMoment = moment(startDate)
+    const endMoment = moment(endDate)
+
+    // This prevents the user from inputting invalid data. Validation is not required since this behaviour
+    // is fully prevented (see also use of minDate and maxDate in DateTimePickers)
+    if (
+      (isStartDate && (startDate === newRangeBoundary.format('YYYYMMDD') || (isStartDate && newRangeBoundary.diff(endMoment) > 0))) ||
+      (!isStartDate && (endDate === newRangeBoundary.format('YYYYMMDD') || newRangeBoundary.diff(startMoment) < 0))
+    ) {
+      return
```

Comment:
Yeah thanks for pointing this out, there's also some other weird behaviour that you can trigger where the exception date is automatically populated, but not respected by the validation. I'm going to address this and those issues in a separate forthcoming PR because I suspect the solution will be similar, and the previous issue existed even in prod